src/inmanta/env.py

```diff
@@ -547,10 +550,14 @@ def create_log_content_files(title: str, files: list[str]) -> list[str]:
                 if "versions have conflicting dependencies" in line:
                     conflicts.append(line)
                 # Get the indexes line from full_output
+                # This is not printed when not using any index or when only using PyPi
                 if "Looking in indexes:" in line:
                     indexes = line
             if not_found:
-                if indexes:
+                no_index: bool = "--no-index" in cmd or strtobool(sub_env.get("PIP_NO_INDEX", "false"))
```

Comment:
1. I now realize that this is actually incorrect. We are not taking into account how the `no-index` setting is configured in the configuration files. If the `--no-index` cli option was not provided and the `PIP_NO_INDEX` environment setting was not set, but the `no-index` config option is set to `true` in the configuraiton file, then error message will say `Packages %s were not found at PyPI.`, while it should say `Packages %s were not found. No indexes were used.`
2. Next to that I think we should make the error message `Packages %s were not found. No indexes were used.` more clear. Currently we only report that pip did not use any indexes, but we leave it entirely to the end-user to figure out why. The error message should hint towards the `no-index` option. Otherwise the end-user will have a hard time figuring out what is going on.

@wouterdb Do you have an opinion on these two points? Because you reviewed the previous PR as well.

Reply:
Indeed but isn't this starting to become a little to "fine-grained"? I mean I liked to idea of helping the user by giving the info about the pip index that is used but I feel there are so much corner cases that we are trying to cover that are starting to add a lot of if conditions which I don't like. 
Can't we just do:
```
if indexes:
    msg = "Packages %s were not found in the given indexes. (%s)" % (", ".join(not_found), indexes)
else:
    msg = "Packages %s were not found. This could be because '--no-index' is set somewhere (config, env_vars) or because only the Pypi Index is used." % (", ".join(not_found))
```
?